_drafts/zio-fibers.md

```diff
@@ -0,0 +1,289 @@
+
+Joining a fiber lets us also gather the result of its execution. In the example, the variable `result` has type `(String, String)`, and we successfully use it in the next step of the routine. We also introduced the `*>` operator, which is an alias for the `zipRight` function, and let us concatenate the execution of two effects not depending on each other.
+
+The execution of the `concurrentWakeUpRoutine` function prints precisely what we expect:
+
+```shell
+[zio-default-async-2]: Going to the bathroom
+[zio-default-async-3]: Boiling some water
+[zio-default-async-6]: (Going to the bathroom,Boiling some water)
+[zio-default-async-6]: (Going to the bathroom,Boiling some water)...done
+[zio-default-async-6]: Preparing the coffee
+```
+
+Bob uses the fiber to go to the bathroom, and the fiber that boils the water runs concurrently on different threads. Then, ZIO executes the fiber used to prepare the coffee in a new thread only after the previous fibers succeeded.
+
+However, this code is not using systems threads directly for concurrent computation. In fact, the ZIO runtime is smart to free up threads when they're not active, so **ZIO fibers don't block any thread during the waiting** associated with the call of the `join` method. Just remember that a `Fiber` represents only a data structure, a blueprint of computation, and not the computation itself.
+
+### 7. Interrupting a Fiber
+
+The last main feature on fibers that the ZIO library provides is interrupting the execution of a fiber. Why should we interrupt a fiber? The main reason is that some action external to the fiber execution turns the fiber useless. So, to not waste system resources, it's better to interrupt the fiber.
+
+Imagine that, while Bob is taking a bath, and the water is waiting to boil, Alice calls him to breakfast in a Cafe. But, then, Bob doesn't need the water anymore. So, we should interrupt the associated fiber.
+
+In ZIO, we can interrupt a `Fiber` using the `interrupt` function:
+
+```scala
+// From ZIO library
+trait Fiber[+E, +A] {
+  def interrupt: UIO[Exit[E, A]]
+}
+```
+
+As we can see, the interruption of fiber results in an effect that always succeeds (`UIO[A]` is just a type alias for `ZIO[Any, Nothing, A]`) with an `Exit` value. If the fiber already succeeded with its value when interrupted, then ZIO returns an instance of `Exit.Success[A]`, an `Exit.Failure[Cause.Interrupt]` otherwise.
+
+So, without further ado, let's model the calling of Alice with an effect:
+
+```scala
+val aliceCalling = ZIO.succeed("Alice's call")
+```
+
+Then, we want to add some delay to the effect associated with the boiling water. So we use the ZIO primitive `ZIO.sleep` to create an effect that waits for a while:
+
+```scala
+val boilingWaterWithSleep =
+  boilingWater.debug(printThread) *>
+    ZIO.sleep(5.seconds) *>
```

Comment:
need to be careful with the duration thing, because it's not the standard Scala package - I'll add it myself

Reply:
Thanks.